diff --git a/package/libs/openssl/Config.in b/package/libs/openssl/Config.in
index 3ad8a66..e0d4d50 100644
--- a/package/libs/openssl/Config.in
+++ b/package/libs/openssl/Config.in
@@ -266,6 +266,13 @@ config OPENSSL_ENGINE_CRYPTO
 		API modules) for /dev/crypto to show up and use hardware
 		acceleration; otherwise it falls back to software.
 
+config OPENSSL_THREADS
+       bool
+       prompt "Threading support"
+       default y
+       select PACKAGE_libpthread
+
+
 config OPENSSL_WITH_ASYNC
 	bool
 	prompt "Enable asynchronous jobs support"
diff --git a/package/libs/openssl/Makefile b/package/libs/openssl/Makefile
index 732e3eb..747fe17 100644
--- a/package/libs/openssl/Makefile
+++ b/package/libs/openssl/Makefile
@@ -33,6 +33,7 @@ PKG_CPE_ID:=cpe:/a:openssl:openssl
 PKG_CONFIG_DEPENDS:= \
 	CONFIG_OPENSSL_ENGINE \
 	CONFIG_OPENSSL_ENGINE_CRYPTO \
+        CONFIG_OPENSSL_THREADS \
 	CONFIG_OPENSSL_NO_DEPRECATED \
 	CONFIG_OPENSSL_OPTIMIZE_SPEED \
 	CONFIG_OPENSSL_PREFER_CHACHA_OVER_GCM \
@@ -254,6 +255,12 @@ else
   OPENSSL_OPTIONS += no-engine
 endif
 
+ifdef CONFIG_OPENSSL_THREADS
+  OPENSSL_OPTIONS += threads
+else
+  OPENSSL_OPTIONS += no-threads
+endif
+
 ifndef CONFIG_OPENSSL_WITH_GOST
   OPENSSL_OPTIONS += no-gost
 endif
